pages/star-search/index.tsx

```diff
@@ -365,33 +357,36 @@ export default function StarSearchPage({ userId, bearerToken, ogImageUrl }: Star
             return;
           }
 
-          let jsonContent: WidgetDefinition;
-          const { result } = matched.groups;
-
           try {
+            let jsonContent: WidgetDefinition;
+            const { result } = matched.groups;
+
+            // deserialize the json dump from the payload
+            const payload = JSON.parse(result) as StarSearchPayload;
+
             // function_call means we're loading a widget definition for the enriched UI
-            if (eventType === "function_call") {
-              jsonContent = JSON.parse(result);
+            if (payload.content.type === "function_call") {
+              jsonContent = JSON.parse(payload.content.parts[0]);
```

Comment:
Is this guaranteed to have at least one item in the array?

Reply:
Yes: but probably better to add a check 👍🏼